redis/src/connection.rs

```diff
@@ -31,9 +31,20 @@ use rustls::OwnedTrustAnchor;
 #[cfg(feature = "tls-rustls-webpki-roots")]
 use webpki_roots::TLS_SERVER_ROOTS;
 
-#[cfg(all(feature = "tls-rustls", not(feature = "tls-rustls-webpki-roots")))]
+#[cfg(all(
+    feature = "tls-rustls",
+    not(feature = "tls-native-tls"),
+    not(feature = "tls-rustls-webpki-roots")
+))]
 use rustls_native_certs::load_native_certs;
 
+#[cfg(feature = "tls-rustls")]
+use crate::tls::TlsConnParams;
+
+#[cfg(not(feature = "tls-rustls"))]
+#[derive(Clone, Debug)]
+pub struct TlsConnParams;
```

Comment:
Maybe add a private field for future proofing ? Together with a "constructor" function, of course.

Reply:
not really sure what you meant here... could you pls jot down some code to exemplify?

Reply:
Adding `#[non_exchaustive]` will also make this type non-construct-able publicly.